### PR TITLE
CP Empty Maps

### PR DESCRIPTION
**Pull request by Maithem**
This patch logically revers PR#808 and adds a new test case
which encodes an edge case where accessing an empty map
doesn't cause a StackOverflowException.

### CODE REVIEW

**Comment by slfritchie** [comment]
I haven't fully caught up yet on mid-week events, sorry.  #820 also looks good to me and should be merged as long as Robert doesn't mind.

File: test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointSmokeTest.java [hunk 1]
```diff
@@ -525,5 +526,48 @@ public void MultiCheckpointWriterTestInner(boolean consecutiveCkpoints) throws E
             assertThat(m2B.get(keyB)).isEqualTo(i);
         }
     }
+
+    @Test
+    public void emptyCheckPoint() throws Exception {
```
Comment (medhavidhawan) [inline test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointSmokeTest.java]: It would be good to write comments.

File: test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointSmokeTest.java [hunk 2]
```diff
@@ -525,5 +526,48 @@ public void MultiCheckpointWriterTestInner(boolean consecutiveCkpoints) throws E
...
+        final String author = "CPWriter";
+        final int iter = 1000;
+
+        for (int x = 0; x < iter; x++) {
+            mA.put(Integer.toString(x), (long) x);
+            mB.put(Integer.toString(x), (long) x);
+        }
+
+        for (String key : mA.keySet()) {
+            mA.remove(key);
```
Comment (medhavidhawan) [inline test/src/test/java/org/corfudb/runtime/checkpoint/CheckpointSmokeTest.java]: Do we need to test the case where put and remove are done in transactions.
